sources/bid/bid.c

```diff
@@ -0,0 +1,156 @@
+
+#include <bid/bid.h>
+#include <bid/c_convert/bid_convert_c.h>
+#include <brutal/base.h>
+#include <brutal/io.h>
+#include <brutal/io/buffer.h>
+#include <brutal/io/scan.h>
+#include <brutal/log.h>
+#include "brutal/alloc/global.h"
+#include "brutal/host/io.h"
+#include "brutal/host/log.h"
+#include "brutal/text/str.h"
+
+size_t get_line_count_from_input(size_t offset, Str input, int *line_offset)
+{
+
+    size_t count = 0;
+    for (size_t i = 0; i < offset; i++)
+    {
+        (*line_offset)++;
+        if (input.buffer[i] == '\n')
+        {
+            (*line_offset) = 0;
+            count++;
+        }
+    }
+    return count;
+}
+
+Str get_line(size_t line, Str input)
+{
+    size_t count = 0;
+    size_t start = 0;
+    size_t end = 0;
+
+    for (size_t i = 0; i < input.len; i++)
+    {
+        if (input.buffer[i] == '\n')
+        {
+            count++;
+            if (count == line && start == 0)
+            {
+                start = i + 1;
+            }
+            else if (start != 0)
+            {
+                end = i;
+                break;
+            }
+        }
+    }
+
+    if (end == 0 && start != 0)
+    {
+        end = input.len;
+    }
+
+    return str_cast_n(end - start, input.buffer + start);
+}
```

Comment:
We should move that into the scan object, like scan_current_location(struct scan* self);